server/app/views/admin/apikeys/ApiKeyIndexView.java

```diff
@@ -148,19 +148,13 @@ private DivTag renderApiKey(
     if (apiKey.isRetired()) {
       statsDiv.with(p("Retired " + dateConverter.formatRfc1123(apiKey.getRetiredTime().get())));
     } else {
-
-      String onSubmitJs =
-          "return confirm('Retiring the API key is permanent and will prevent"
-              + " anyone from being able to call the API with the key. Are you"
-              + " sure you want to retire "
-              + apiKey.getName()
-              + "?')";
       linksDiv.with(
           form(makeCsrfTokenInputTag(request))
               .withAction(controllers.admin.routes.AdminApiKeysController.retire(apiKey.id).url())
               .withMethod("POST")
-              .withOnsubmit(onSubmitJs)
               .withId(String.format("retire-%s", keyNameSlugified))
+              .withData("api-key-name", apiKey.getName())
+              .withClasses("retire-key-form")
```

Comment:
question/suggestion (non-blocking): I would prefer for this to be an id since we use it to reference one element, but if there's a reason you prefer a class I'm open to it.

Reply:
Hmm there can be multiple API keys on the page and thus multiple retire key buttons. That's why the ID on line 155 has the key name in it (which actually doesn't have to be unique, but that's off topic).

Reply:
Oh that makes sense! In the past I've seen people use [reference classes](https://github.com/civiform/civiform/blob/main/server/app/views/style/ReferenceClasses.java) for this so it's clear it's not a class to control styling.

Reply:
Yeah I'm aware of ReferenceClasses but have never been sure when to/not to use it, and I've seen plenty of examples not using it... so I guessed that maybe it was if the string was re-used a lot within Java, since you can't reference it in JS. I think I'll leave it, but maybe we should clarify the best practices for ReferenceClasses.

Reply:
That works for me! And agreed about clarifying best practices for Reference Classes.